modules/jarm/scanner.go

```diff
@@ -120,20 +114,14 @@ func (scanner *Scanner) Scan(target zgrab2.ScanTarget) (zgrab2.ScanStatus, inter
 			return zgrab2.TryGetScanStatus(err), nil, err
 		}
 
-		_, err = conn.Write([]byte(data))
+		_, err = conn.Write(jarm.BuildProbe(probe))
 		if err != nil {
 			rawhashes = append(rawhashes, "")
 			conn.Close()
 			continue
 		}
 
-		// ret, err = zgrab2.ReadAvailable(conn)
-		ret, err = zgrab2.ReadAvailableWithOptions(conn, 1484, 500*time.Millisecond, 0, 1484)
-		if err != io.EOF && err != nil {
-			rawhashes = append(rawhashes, "")
-			conn.Close()
-			continue
-		}
+		ret, _ = zgrab2.ReadAvailableWithOptions(conn, 1484, 500*time.Millisecond, 0, 1484)
```

Comment:
It appears to be the case that error responses from the target should still be processed by `ParseServerHello` which contains logic for handling the various failure responses appropriately.